pkg/controller/nodeipam/ipam/cloud_cidr_allocator.go

```diff
@@ -328,9 +328,17 @@ func (ca *cloudCIDRAllocator) updateCIDRAllocation(nodeName string) error {
 	// nodes in clusters WITHOUT multi-networking are expected to have
 	// only 1 network-interface and 1 alias IPv4 range or/and 1 IPv6 address
 	// multi-network cluster may have 1 interface with multiple alias
-	if len(instance.NetworkInterfaces) == 1 &&
+	multiNetworkingDisabled := len(instance.NetworkInterfaces) == 1 &&
 		(len(instance.NetworkInterfaces[0].AliasIpRanges) == 1 ||
-			ca.cloud.GetIPV6Address(instance.NetworkInterfaces[0]) != nil) {
+			ca.cloud.GetIPV6Address(instance.NetworkInterfaces[0]) != nil)
```

Comment:
This condition to check for IPv6 address is wrong. It causes single stack IPv4 clusters to also get dual stack pod CIDRs if the subnet happens to be dual stack. See https://buganizer.corp.google.com/issues/315684769

We should revert https://github.com/kubernetes/cloud-provider-gcp/commit/ce1a81aeb805a9d69562c8f0bde819f459bb182f?diff=unified&w=0 to some extent here.

Reply:
Thanks for pointing this out! I reverted the condition check and the unit test, please review.

Reply:
yep